### DISCUSSION TITLE
How to add content to a Tab

### DISCUSSION THREAD

**Discussion by cristianprice** [Textualize/textual#2942] [Q&A] [answered]
Hi,
First of all I have to say this: you have my gratitude for this wonderful project. Now to the point. I am curious how do you add content to a tab ? I could not find any way in doing this ? Am I getting the wrong idea? Arent tabs supposed to have content  ?
Thanks,
C

**Comment by TomJGooding** [comment, marked as answer]
You might want to look at the [TabbedContent](https://textual.textualize.io/widgets/tabbed_content/) widget, which combines the [Tabs](https://textual.textualize.io/widgets/tabs/) and [ContentSwitcher](https://textual.textualize.io/api/content_switcher/) widgets.

Perhaps the `Tabs` docs should be updated to make this clearer?

**Reply by cristianprice** [reply]
Yes. I have discovered this today. Thanks. Then, no offence, but what is the use case for Tabs ? How would you use it ? If feels a bit confusing.
Thanks and have an awesome day.

**Reply by TomJGooding** [reply]
I'm not a maintainer of Textual just an ~~annoying hanger-on~~ eager contributor, just to confrm. :)

The `Tabs` widget I think came a little while before the `TabbedContent` for starters. But I suppose there might also be times where the content is not mutually exclusive so `TabbedContent` would be overkill.

**Reply by davep** [reply]
> but what is the use case for Tabs

A few widgets in Textual are built of more "atomic" widgets, and we generally don't hide them away as they will have utility for developers who may want to build their own versions of the widgets that might get built atop of them. `TabbedContent` is a great example, in that it takes `Tabs` (which in turn builds on `Tab`) and combines it with `ContentSwitcher` to make a tabbed content switcher. Developers may want to take `Tabs` and combine it with other things for a very different take.

**Reply by TomJGooding** [reply]
I forgot that I actually have a use case in one of the first apps I built with Textual!

Selecting a tab could for example simply scroll to the relevant content rather than switch it:

![image](https://github.com/Textualize/textual/assets/101601846/b49e35a8-d856-42c9-95ce-52ce3e2906f3)

**Reply by cristianprice** [reply]
That makes sense. Thanks for the answers.
